### DISCUSSION TITLE
How do I process the physics system in single thread?

### DISCUSSION THREAD

**Discussion by AndreaCatania** [jrouwe/JoltPhysics#587] [Q&A] [answered]
Hello, I need to make sure that the physics engine is executed in single thread and at the end of the update function: the physics system is not running any longer on any thread.

```cpp
	void update()
	{
		const int cCollisionSteps = 1;
		const int cIntegrationSubSteps = 1;
		physics_system.Update(cDeltaTime, cCollisionSteps, cIntegrationSubSteps, &temp_allocator, &job_system);
		// WHAT TO PUT HERE TO WAIT THE PHYSICS_SYSTEM TO FINISH
	}
```
How can I assure the process runs in single thread?

**Comment by jrouwe** [comment, marked as answer]
The PhysicsSystem::Update function internally spawns jobs AND waits for them to complete. So the work is done when the function returns.

**Reply by AndreaCatania** [reply]
Nice, thanks for the confirmation 👌
